src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -389,14 +414,12 @@ ZNSSegmentManager::open_ertr::future<SegmentRef> ZNSSegmentManager::open(
   LOG_PREFIX(ZNSSegmentManager::open);
   return seastar::do_with(
     blk_zone_range{},
-    [=] (auto &range){
+    [=] (auto &range) {
       range = make_range(
-	id, 
-	metadata.zone_size, 
-	metadata.block_size, 
-	metadata.first_segment_offset);
```

Comment:
Ok

Reply:
I assumed that for ZNS SSDs first zone would be needed for the super block, which made the first_segment_offset obsolete since it would be always equal to zone size.
But I guess we could handle the first zone as a special case and in that scenario first_segment_offset would make sense. Will keep it.

Reply:
I think either dropping `first_segment_offset` from `zns_sm_metadata_t`, or using it in `make_range()` is good, but not to leave it alone without any user.